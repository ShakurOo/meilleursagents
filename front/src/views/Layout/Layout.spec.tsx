import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Layout } from './';

describe('Layout component', () => {
  it('does match snapshots and styles', async () => {
    const { container } = render(<Layout />, { wrapper: BrowserRouter });
    expect(container).toMatchSnapshot();
  });

  it('does display <Header />', async () => {
    render(<Layout />, { wrapper: BrowserRouter });
    expect(screen.getByTestId('main-header')).toBeInTheDocument();
  });
});
