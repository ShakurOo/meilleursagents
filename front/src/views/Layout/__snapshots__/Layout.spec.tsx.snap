// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component does match snapshots and styles 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1tzvssz-MuiPaper-root-MuiAppBar-root"
    data-testid="main-header"
  >
    <div
      class="MuiContainer-root css-ur2jdm-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <div
          class="css-1hbwp7i"
        >
          <svg
            id="svg2"
            version="1.1"
            viewBox="0 0 654.9 183.5"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g
              id="g16"
            >
              <path
                d="M157,156.4c35.4-35.6,35.4-93.6,0.1-129.3C139.7,9.6,116.6,0,91.8,0C67,0,43.8,9.6,26.5,27.1
        C-0.7,54.7-7.7,96.6,9,131.5c2,4.1,6.1,6.7,10.7,6.7c3.2,0,6.2-1.3,8.4-3.5c3.5-3.6,4.5-9,2.3-13.4C17.8,94.9,23.3,63.4,44.2,43
        C57,30.5,73.9,23.6,91.8,23.6c17.1,0,33.4,6.4,46,17.9c13.8,12.7,21.6,29.9,22,48.6c0.4,18.7-6.6,36.2-19.7,49.5
        c-12.9,13-30,20.1-48.3,20.1c-9.8,0-19.6-2.2-28.5-6.2c-1.5-0.7-3.1-1-4.8-1c-3.1,0-6.2,1.3-8.5,3.5c-2.8,2.8-4,6.6-3.2,10.5
        c0.7,3.9,3.2,7.1,6.9,8.8c11.8,5.4,25,8.2,38.1,8.2C116.5,183.5,139.6,173.9,157,156.4L157,156.4z M65,76.1l26.7-21.2l26.7,21.2
        l0,44.1H65V76.1"
                fill="#9c27b0"
              />
              <path
                d="M296.9,48.2v26.7h-13.5V49c0-8-4-12.3-11.1-12.3c-3.1,0-6.7,1.3-9.8,5.7c0.3,1.9,0.5,3.7,0.5,5.8
        v26.7h-13.5V49c0-8-4-12.3-11.1-12.3c-2.9,0-6.3,1.6-9.3,5.7v32.5h-13.5V25.7H229v3.1c3-2.8,6.3-4.3,12.2-4.3
        c6.2,0,11.8,2.5,15.8,6.8c5.3-4.7,10.1-6.8,18.2-6.8C287.2,24.4,296.9,33.7,296.9,48.2"
                fill="#9c27b0"
                id="path24"
              />
              <path
                d="M321.6,45.1H344c-0.7-6.3-4.7-9.2-11-9.2C326.3,35.9,322.8,39.9,321.6,45.1z M357.4,54.2h-36.1
        c1,5.7,4.7,10,12.5,10c3.6,0,6.1-0.6,7.8-2.8h15c-3.3,9.3-12.1,14.6-22.9,14.6c-15.8,0-26.2-11.1-26.2-25.8
        c0-14.6,10.3-25.9,25.4-25.9c14.3,0,24.9,9.5,24.9,24.1C357.7,50.6,357.7,52.2,357.4,54.2"
                fill="#9c27b0"
              />
              <path
                d="M382.4,25.7v49.2h-13.5V25.7H382.4z M367.3,8.1c0-4.6,3.6-8,8.3-8c4.5,0,8.3,3.4,8.3,8
        c0,4.5-3.7,7.9-8.3,7.9C370.9,16.1,367.3,12.7,367.3,8.1"
                fill="#9c27b0"
              />
              <path
                d="M395.8,74.8h13.5V1.6h-13.5V74.8z"
                fill="#9c27b0"
              />
              <path
                d="M422.2,74.9h13.5V1.6h-13.5V74.9z"
                fill="#9c27b0"
              />
              <path
                d="M460.8,44.9h22.4c-0.7-6.3-4.7-9.2-11-9.2C465.5,35.8,462,39.8,460.8,44.9z M496.6,54.1h-36.1
        c1,5.7,4.7,10,12.5,10c3.6,0,6.1-0.6,7.8-2.8h15c-3.3,9.3-12.1,14.6-22.9,14.6c-15.8,0-26.2-11.1-26.2-25.8
        c0-14.6,10.3-25.9,25.4-25.9c14.3,0,24.9,9.5,24.9,24.1C496.9,50.5,496.9,52.1,496.6,54.1"
                fill="#9c27b0"
              />
              <path
                d="M556.7,25.7v49.2h-13.5V71c-4,3.2-8.7,5.2-14.2,5.2c-13.1,0-21.8-9.6-21.8-24V25.7h13.5v25.6
        c0,8,4,12.6,11.1,12.6c3.7,0,7.8-2.3,11.3-6.5V25.7H556.7"
                fill="#9c27b0"
              />
              <path
                d="M602.2,24.7v12.8c-7.6,0.4-13.2,2.7-17.4,7.4v29.8h-13.5V25.6h13.5v5.7
        C589.2,27,594.9,24.7,602.2,24.7"
                fill="#9c27b0"
              />
              <path
                d="M251.8,141.7v-15.6c-3.2-4.5-7.6-5.8-12.1-5.8c-7.7,0-13.1,6.2-13.1,13.6
        c0,7.3,5.4,13.5,13.1,13.5C244.2,147.5,248.6,146.3,251.8,141.7L251.8,141.7z M265.3,109.3v49.2h-13.5v-2.8c-3.8,2.5-8.7,4-14.1,4
        c-13.6,0-24.6-11.6-24.6-25.8c0-14.3,10.9-25.9,24.6-25.9c5.5,0,10.3,1.5,14.1,4v-2.8L265.3,109.3"
                fill="#9c27b0"
              />
              <path
                d="M313.9,138.6V126c-3.2-4.6-7.6-6.2-12.1-6.2c-7.7,0-13.1,5.2-13.1,13c0,5.8,5.4,12.1,13.1,12.1
        C306.3,144.8,310.7,143.3,313.9,138.6L313.9,138.6z M327.4,109.3v49.5c0,15.4-10.5,24.7-25.2,24.7c-14.1,0-24.7-8.8-25.7-21.1h13.7
        c0.6,5.3,5,8.9,12,8.9c6.6,0,11.7-4.3,11.7-12.4v-6.3c-3.8,2.5-8.7,3.9-14.1,3.9c-13.6,0-24.6-11-24.6-23.7
        c0-14.8,10.9-24.7,24.6-24.7c5.5,0,10.3,1.4,14.1,3.9v-2.7L327.4,109.3"
                fill="#9c27b0"
              />
              <path
                d="M351.9,128.7h22.4c-0.7-6.3-4.7-9.2-11-9.2C356.7,119.6,353.2,123.6,351.9,128.7L351.9,128.7z
        M387.7,137.9h-36.1c1,5.7,4.7,10,12.5,10c3.6,0,6.1-0.6,7.8-2.8h15c-3.3,9.3-12.1,14.6-22.9,14.6c-15.8,0-26.2-11.1-26.2-25.8
        c0-14.6,10.3-25.9,25.4-25.9c14.3,0,24.9,9.5,24.9,24.1C388,134.3,388,135.9,387.7,137.9"
                fill="#9c27b0"
              />
              <path
                d="M447.2,131.8v26.7h-13.5v-25.9c0-8-4-12.3-11.1-12.3c-3.7,0-7.8,2.2-11.3,6.3v31.9h-13.5v-49.2
        h13.5v3.7c4-3.2,8.7-5,14.2-5C438.6,108.1,447.2,117.4,447.2,131.8"
                fill="#9c27b0"
              />
              <path
                d="M497.1,157.6c-4.5,1.8-8.3,2.2-11.9,2.2c-13.2,0-20.6-8.3-20.6-20.8v-18.3H454v-11.3h10.5V91.5
        h13.5v17.8h18.1v11.3h-18.1V139c0,5.8,3.6,8.4,8.7,8.4c3.1,0,6-0.9,8.5-2L497.1,157.6"
                fill="#9c27b0"
                id="path48"
              />
              <path
                d="M528,149.5c5.2,0,8.8-2,8.8-4.8c0-2-1.7-3.6-5.1-4.2l-10.6-1.9c-7.2-1.2-14.5-5.2-14.5-14.4
        c0-10.1,8.6-16.2,21-16.2c8.9,0,20.6,2.5,21.8,16.6h-12.5c-0.3-4.4-4.5-6.4-9.3-6.4c-5.2,0-7.6,2.3-7.6,4.6c0,1.9,1.3,4,6.1,4.8
        l8.9,1.4c7.6,1.3,14.8,4.8,14.8,14.3c0,10.9-9.5,16.3-22,16.3c-10.8,0-21.8-4-22.8-18.4l12.5,0.1
        C518.2,147.1,522.2,149.5,528,149.5"
                fill="#9c27b0"
              />
              <path
                d="M633.2,65.9c5.2,0,8.8-2,8.8-4.8c0-2-1.6-3.6-5.1-4.2L626.3,55c-7.2-1.2-14.5-5.2-14.5-14.4
        c0-10.1,8.6-16.2,21-16.2c8.9,0,20.6,2.5,21.8,16.6h-12.5c-0.3-4.4-4.5-6.4-9.3-6.4c-5.2,0-7.6,2.3-7.6,4.7c0,1.9,1.3,4,6.1,4.8
        l8.9,1.4c7.6,1.3,14.7,4.8,14.7,14.3c0,10.9-9.5,16.3-22,16.3c-10.8,0-21.8-4-22.8-18.4l12.5,0.1C623.4,63.4,627.5,65.9,633.2,65.9
        "
                fill="#9c27b0"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-xia04d"
        >
          <button
            class="MuiLoadingButton-root MuiLoadingButton-loading MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled unread-counter css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            disabled=""
            id="mui-1"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiLoadingButton-startIconLoadingCenter MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
            </span>
            <div
              class="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorCenter css-1yt7yx7-MuiLoadingButton-loadingIndicator"
            >
              <span
                aria-labelledby="mui-1"
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
                role="progressbar"
                style="width: 16px; height: 16px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
          </button>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth realtors-select css-q8hpuo-MuiFormControl-root"
          >
            <button
              class="MuiLoadingButton-root MuiLoadingButton-loading MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled css-6m3mwv-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              disabled=""
              id="mui-2"
              tabindex="-1"
              type="button"
            >
              <div
                class="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorCenter css-1yt7yx7-MuiLoadingButton-loadingIndicator"
              >
                <span
                  aria-labelledby="mui-2"
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 16px; height: 16px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
              ...
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="css-1ata9nr"
  />
</div>
`;
